lib/views/chatroom/chatroom_components/chat_bar.dart

```diff
@@ -0,0 +1,262 @@
+import 'package:custom_pop_up_menu/custom_pop_up_menu.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:group_chat_example/constants.dart';
+import 'package:group_chat_example/utils/ui_utils.dart';
+import 'package:group_chat_example/views/chatroom/views/chatroom_polls_page.dart';
+
+class ChatBar extends StatefulWidget {
+  const ChatBar({super.key});
+
+  @override
+  State<ChatBar> createState() => _ChatBarState();
+}
+
+class _ChatBarState extends State<ChatBar> {
+  late CustomPopupMenuController _popupMenuController;
```

Comment:
It is better to mark these variables as final.

Reply:
final will require me to initialise it there and then, I want it to be initialised with the state.

Reply:
Actually no, we are using the `late` keyword to tell dart that the variable will be initialized afterwards. It will not throw an error if we mark the variable as `late final` and initialize it at `initState`.